src/components/Table/Table.jsx

```diff
@@ -27,41 +27,79 @@ const cellProps = (
 
 function Table(props) {
   const { currentSort, fields, data, ariaLabelledBy } = props;
+  const [sortDirection, setSortDirection] = useState(currentSort?.order);
+  let sortLabel;
+  let rowData;
+
+  useEffect(() => {
+    rowData = sortData();
+    sortLabel = getSortLabel();
+  });
+
+  const sortData = () => {
+    if (sortDirection === 'ASC') {
+      return data.sort((a, b) => {
+        return a[currentSort.value] > b[currentSort.value] ? 1 : -1;
+      });
+    } else if (sortDirection === 'DESC') {
+      return data.sort((a, b) => {
+        return a[currentSort.value] < b[currentSort.value] ? 1 : -1;
+      });
+    }
+  };
+
+  const getSortLabel = () =>
+    sortDirection === 'ASC' ? 'ascending' : 'descending';
+
+  sortLabel = getSortLabel();
+  rowData = sortDirection ? sortData() : data;
 
   return (
     <table aria-labelledby={ariaLabelledBy} className="responsive" role="table">
       <thead>
         <tr role="row">
           {fields.map(field =>
-            field.nonSortable ? (
-              <th key={field.value}>{field.label}</th>
-            ) : (
-              <th
-                key={field.value}
-                className={borderClasses}
-                role="columnheader"
-                scope="col"
-              >
-                <button className="va-button-link vads-u-font-weight--bold vads-u-color--base vads-u-text-decoration--none">
+            field.sortable && field.value == currentSort?.value ? (
+              <th key={field.value} role="columnheader" scope="col">
+                <button
+                  className="va-button-link vads-u-font-weight--bold vads-u-color--base vads-u-text-decoration--none sort"
+                  onClick={() =>
+                    setSortDirection(sortDirection === 'ASC' ? 'DESC' : 'ASC')
```

Comment:
It needs to be the way it is--`setSortDirection(sortDirection === 'ASC' ? 'DESC' : 'ASC')`--because we're setting `sortDirection` to the opposite of whatever it is when the user clicks the sort button.

Also, if no column is set to be sorted, then `currentSort.order` is `undefined`. If it's `undefined`, then we don't sort any of the data. So that also need to stay the way it is, with the current implementation.

Reply:
Ahh, the opposite makes sense. Somehow I overlooked that.